### PR TITLE
Fix input box validation

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/13667

The logic to add a custom input validation message is relatively new - added in https://github.com/microsoft/azuredatastudio/pull/8909. But it appears that due to race condition issues I fixed in https://github.com/microsoft/azuredatastudio/pull/13317 we never noticed any problems.

The root problem here is that we were manually calling showMessage while doing our validate - but because this was happening after the call to inputbox.validate then if we were showing an error initially (a common thing to happen since values aren't initialized until later) when the validation became successful the error message was never hidden properly.

Instead of doing this ourselves we should just be relying on the base class to handle all that for us - and use the validate callback that we pass in to tell it whether or not to show the message. 

While doing this I also noticed another issue where for input boxes using default validation (i.e. number inputs checking min/max) we weren't setting hideErrors to false until AFTER doing the validate - which meant that if someone typed in an invalid number initially it wouldn't show an error - that would only happen for any subsequent characters being inserted. We now set it to false as soon as there's any changes before calling validate so that any errors are correctly displayed. 

**IMPORTANT** - Currently the behavior is changed for default error messages such as min/max values and required values not being empty. Previously we would display the custom `validationErrorMessage` defined by the extension - but now with my changes we will ALWAYS use the error message that's coming from the input box (if one exists).

To me I think this makes sense - I don't really see the need for a custom message in those cases so using the default one seems correct. And in fact I think the messages are even improved since previously we'd ALWAYS show the same error message, but now we'll show different ones based on whether the field fails custom validation OR is empty when it's required. A good example of this is the Import dialog where it checks that the file exists in the validation : 

Import_Validation

Here's some examples I used to verify functionality : 

Deploy DacPac wizard : 
Deploy_Validation

Create DB Proj wizard : 
DbProj_Validation

Arc PG Dashboard:
Dashboard_Validation

Connection Dialog:
ConnectionDialog_Validation

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
And re: the changes, that means in some places where we use `validationErrorMessage` without a validation function (such as the DB Proj wizard) there's no real point in having that property anymore since it'll never be used. If we agree with this change I'll go through and clean those up.

File: src/sql/workbench/browser/modelComponents/inputbox.component.ts [hunk 1]
```diff
@@ -159,17 +157,14 @@ export default class InputBoxComponent extends ComponentBase<azdata.InputBoxProp
 	}
 
 	public async validate(): Promise<boolean> {
-		let valid = await super.validate();
-		const otherErrorMsg = valid || this.inputElement.value === '' ? undefined : this.validationErrorMessage;
-		valid = valid && this.inputElement.validate();
+		await super.validate();
```
Comment (kisantia) [inline src/sql/workbench/browser/modelComponents/inputbox.component.ts:160]: the result of super.validate() doesn't matter?

**Review by kisantia — APPROVED** [review]
thanks for fixing this! Sounds good to clean up the places without a validation function

File: src/sql/workbench/browser/modelComponents/inputbox.component.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/modelComponents/inputbox.component.ts:160]: Nope, the validate callback already uses this.valid (which is set by super.validate()) so using it here is redundant

**Review by aasimkhan30 — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

**Comment by Charles-Gagnon** [comment]
I'll follow up with a separate PR to remove the unnecessary validation messages

**Pull request merged**